src/pages/Scholarships.jsx

```diff
@@ -45,10 +45,27 @@ function ScholarshipsPage() {
     });
   };
 
-  const pruneQueryParam = (index) => {
-    delete params[index];
-    history.replace({ search: queryString.stringify(params) });
-  };
+  const DEFAULT_SORT_FIELD = 'deadline';
+  const DEFAULT_SORT_DIR = 'asc';
+  let sortField = params.sortField ?? DEFAULT_SORT_FIELD;
+  let sortDir = params.sortDir ?? DEFAULT_SORT_DIR;
+
+  const validateField =
+    sortField === 'deadline' ||
+    sortField === 'amount.min' ||
+    sortField === 'amount.max';
+  if (!validateField) {
+    pruneQueryParam('sortField');
+    sortField = DEFAULT_SORT_FIELD;
+  }
+
+  const validateDir = sortDir === 'asc' || sortDir === 'desc';
+  if (!validateDir) {
+    pruneQueryParam('sortDir');
+    sortDir = DEFAULT_SORT_DIR;
+  }
```

Comment:
This is not exactly what I meant by my previous [comment](https://github.com/beyondhb1079/s4us/pull/379#discussion_r575939396). `pruneQueryParam` calls `history.replace()` which will refresh this component. So, as my [other comment](https://github.com/beyondhb1079/s4us/pull/379#discussion_r578960875) was trying to show, you can simplify this to:
```suggestion
  const sortField = params.sortField ?? DEFAULT_SORT_FIELD;
  const sortDir = params.sortDir ?? DEFAULT_SORT_DIR;
  
  // setSort function (if you still keep it)
  
  if (sortField !== 'deadline' && sortField !== 'amount.min' && sortField !== 'amount.max') {
    pruneQueryParam('sortField');  // prunes the query parameter, which ends up refreshing the component
  }
  // From this point on, sortField is valid.
  if (sortDir !== 'asc' && sortDir != 'desc') {
    pruneQueryParam('sortDir');  // prunes the query parameter, which ends up refreshing the component
  }
  // From this point on, sortDir is valid.
```